packages/core-data/src/selectors.js

```diff
@@ -238,3 +238,16 @@ export function isPreviewEmbedFallback( state, url ) {
 	}
 	return preview.html === oEmbedLinkCheck;
 }
+
+/*
+ * Return whether the user has media upload permissions.
+ *
+ * @return {boolean} Does the user have media upload permissions?
+ */
+export function hasUploadPermissions() {
+	// Hardcode hasUploadPermissions to true for now - in future this value will be
+	// correctly populated from a REST API call that will store user permissions
+	// See github issue for more information:
+	// https://github.com/WordPress/gutenberg/issues/3672
+	return true;
```

Comment:
This is where I've hard-coded hasUploadPermissions to true. Setting this to false simulates what it's like for a user without permissions.